curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c3.md

```diff
@@ -0,0 +1,136 @@
+---
+id: 67116d7584d0b469b14579c3
+title: Step 6
+challengeType: 1
+dashedName: step-6
+---
+
+# --description--
+
+Another useful thing to do is to get the books by their author.
+
+Create a `getBooksByAuthor` function with a `catalog` and `author` parameters.
+
+The function must return an array that contains the books by a particular author.
+
+# --hints--
+
+You should create a `getBooksByAuthor` function.
+
+```js
+assert.isFunction(getBooksByAuthor)
+```
+
+Your `getBooksByAuthor` function should have a `catalog` and `author` parameters.
+
+```js
+assert.match(getBooksByAuthor.toString(), /catalog,\s*author|author,\s*catalog/)
+```
+
+Your `getBooksByAuthor` function should return an array.
+
+```js
+assert.isArray(getBooksByAuthor(library))
+```
+
+Your `getBooksByAuthor` function should use an array method.
```

Comment:
same comment here as earlier. This hint will need to be more specific since we are looking for high order functions instead of any array method like push or pop